src/cargo/core/compiler/context/mod.rs

```diff
@@ -396,6 +426,18 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
         self.files.as_mut().unwrap()
     }
 
+    /// Return the list of filenames read by cargo to generate the Context
+    /// (all Cargo.toml, etc).
+    pub fn inputs(&self) -> CargoResult<Vec<PathBuf>> {
+        let mut inputs = Vec::new();
+        for id in self.packages.package_ids() {
```

Comment:
I think this can run the risk of reading too much in the sense that it traverses target-specific dependencies and optional ones that aren't activated. Perhaps this could be built up as it goes along?

I also wonder, perhaps each step in the plan should depend on the manifest but not all at once? (you wouldn't want a manifest change to recompile everything, right?)

Reply:
Hmm, it looks like all the Cargo.toml files reported by this function are actually read by cargo during execution (according to strace). Why would we not want to list them as inputs here?

These inputs are just used to determine when we need to re-execute 'cargo --build-plan'; changing one of these inputs won't force the end-user build system to recompile everything.

I might be misunderstanding you here - perhaps we can schedule some time to chat about it.

Reply:
Oh I see, nah if this is just rerunning the build plan that makes sense. Although I think technically the build plan can change without `Cargo.toml` being modified. For example if I move `src/lib.rs` to `src/main.rs` that implicitly removes a library from a crate and crates a binary, although it may not be worthwhile to have that much fidelity?